pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -2436,3 +2436,40 @@ def test_dt64arr_addsub_object_dtype_2d():
 
     assert result2.shape == (4, 1)
     assert all(td._value == 0 for td in result2.ravel())
+
+
+def test_concat_float_datetime64(using_array_manager):
```

Comment:
Could you move this test to `pandas/tests/reshape/concat/test_datetimes.py`? After that this looks good